cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaAdminTopicConfigProvider.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2021 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.config;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import com.linkedin.kafka.cruisecontrol.monitor.LoadMonitor;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.ConfigEntry;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.config.ConfigResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.linkedin.cruisecontrol.common.utils.Utils.validateNotNull;
+
+
+/**
+ * The Kafka topic config provider implementation based on using the Kafka Admin Client for topic level configurations
+ * and files for cluster level configurations. The format of the file is JSON, listing properties:
+ * <pre>
+ *   {
+ *     "min.insync.replicas": 1,
+ *     "an.example.cluster.config": false
+ *   }
+ * </pre>
+ *
+ */
+public class KafkaAdminTopicConfigProvider extends JsonFileTopicConfigProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaAdminTopicConfigProvider.class);
+
+  public static final String CLUSTER_CONFIGS_FILE = "cluster.configs.file";
```

Comment:
Nit: Should we move this to `JsonFileTopicConfigProvider`?

Reply:
Good point, I will move it there and remove from both implementations.